### DISCUSSION TITLE
Instantiate parent model in a discriminated union without specifying `__root__`

### DISCUSSION THREAD

**Discussion by antonymilne** [pydantic/pydantic#5091] [Question]
Related, but not quite the same: #4950, #3876

Given [this code from the docs](https://docs.pydantic.dev/usage/types/#nested-discriminated-unions):
```
class Cat(BaseModel):
    pet_type: Literal['cat']
    meows: int

class Dog(BaseModel):
    pet_type: Literal['dog']
    barks: float

class Pet(BaseModel):
   __root__: Annotated[Union[Cat, Dog], Field(discriminator='pet_type')]
```

I would like to be able to make a `Dog` by instantiating `Pet` (not `Dog`), i.e. do `Pet(pet_type='dog', barks=3.14)`.

Currently I can do either of these:
```
Pet(__root__={'pet_type': 'dog', 'barks': 3.14})
Pet.parse_obj({'pet_type': 'dog', 'barks': 3.14})
```
... but it's not quite what I want. I'd like to do exactly `Pet(pet_type='dog', barks=3.14)`.

Is there any way to achieve this, even if it's a bit hacky? Maybe make `pet_type` a field in `Pet` and then some sort of `@root_validator` to populate the other fields depending on the value of `pet_type`? Will this be possible in Pydantic 2, where I see that `__root__` is disappearing and `discriminator` is changing getting more powerful (#4675)? Or have I just got completely the wrong model structure here and should be doing something else instead? 

Thank you very much! 🙏

**Comment by maxupp** [comment]
Did you ever figure this out? I'm in the same situation, where I want to validate an object against any of the child models of a base model.

**Reply by antonymilne** [reply]
No, I never found a way to do it, sorry. Possibly it's possible with Pydantic 2 though - I didn't investigate that.

**Reply by maxupp** [reply]
There is indeed:
```
MessageModel = Annotated[
    Union[InitMessage, ClientTurn, EndConversation, RateTurn, ResetMessage, CancelTurn], Field(discriminator='type')]


def validate_obj(message):
    return parse_obj_as(MessageModel, message)
```

**Reply by antonymilne** [reply]
There's still not a way to do it just as `MessageModel(...)` directly though?
